scripts/ai/AIDriveStrategyFieldWorkCourse.lua

```diff
@@ -818,6 +818,176 @@ end
 function AIDriveStrategyFieldWorkCourse:getFieldWorkProximity(node)
     return self.fieldWorkerProximityController:getFieldWorkProximity(node)
 end
+
+-----------------------------------------------------------------------------------------------------------------------
+--- Waiting for refilling on the field
+-----------------------------------------------------------------------------------------------------------------------
+
+function AIDriveStrategyFieldWorkCourse:prepareFilling()

```

Comment:
This is a lot of code in these *Filling() functions, not really related to the strategy, maybe move some to ImplementUtil or similar, or its own helper, keeping only the actions/state changes here?